gamification/src/main/java/com/appcoins/wallet/gamification/repository/BdsPromotionsRepository.kt

```diff
@@ -281,4 +281,12 @@ class BdsPromotionsRepository @Inject constructor(private val api: GamificationA
         throwable is UnknownHostException
   }
 
+  override fun getVipReferral(wallet: String): Observable<VipReferralResponse> {
+    return api.getVipReferral(wallet)
+      .toObservable()
+      .onErrorReturn {
+        VipReferralResponse.invalidReferral
```

Comment:
Any error with this service will be treated as an invalid referral. So that the vip referral button is not shown in the promotions page but the promotions page will always work normally